### PR TITLE
Taskbar: Arbitrary QuickLaunch entries

### PR DESCRIPTION
**Pull request by sppmacd**
This PR adds support for arbitrary QuickLaunch entries to Taskbar. This is made by abstracting an entry into a QuickLaunchEntry class.

It's not ideal as doesn't work for some corner-cases (especially when there are e.g. `test=test` and `testtest` files).

We get also one wrapper in Core::System (disown) and nullptr dereference fix when trying to open binary files with Core::ConfigFile.

https://user-images.githubusercontent.com/42967349/147684566-41ce9363-8e53-44fa-9509-71c6ba2dd9b4.mp4

### CODE REVIEW

File: Userland/Libraries/LibCore/ConfigFile.cpp [hunk 1]
```diff
@@ -76,6 +76,12 @@ void ConfigFile::reparse()
 
     while (m_file->can_read_line()) {
         auto line = m_file->read_line();
+
+        if (line.is_null()) {
```
Comment (awesomekling) [inline Userland/Libraries/LibCore/ConfigFile.cpp:80]: The commit message mentions "null lines"
What exactly is a "null line"?

Comment (sppmacd) [reply Userland/Libraries/LibCore/ConfigFile.cpp:80]: Core::File::read_line() sometimes returns a null string (e.g. when there is no line to read, like in binary files, and it seems that not only if can_read_line() is false).
"Null lines" are these lines.
